modules/fedimint-ln/src/db.rs

```diff
@@ -121,140 +121,221 @@ impl_db_lookup!(
+        dbtx.insert_new_entry(&ContractKey(contract_id), &acct)
+            .await
+            .expect("Error inserting ContractAccount");
+
+        let threshold_key = threshold_crypto::PublicKey::from(G1Projective::identity());
+        let preimage = Preimage([
+            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8,
+            9, 0, 1,
+        ]);
+        let offer = IncomingContractOffer {
+            amount: fedimint_core::Amount { msats: 1000 },
+            hash: secp256k1::hashes::sha256::Hash::hash("01234567".as_bytes()),
+            encrypted_preimage: EncryptedPreimage::new(preimage, &threshold_key),
+            expiry_time: None,
+        };
+        let offer_key = OfferKey(offer.hash);
+        dbtx.insert_new_entry(&offer_key, &offer)
+            .await
+            .expect("Error inserting Offer");
+
+        let contract_update_key = ContractUpdateKey(OutPoint {
+            txid: TransactionId::all_zeros(),
+            out_idx: 0,
+        });
+        let lightning_output_outcome = LightningOutputOutcome::Offer {
+            id: OfferId::from_str(
+                "0123456789012345678901234567890101234567890123456789012345678901",
+            )
+            .unwrap(),
+        };
+        dbtx.insert_new_entry(&contract_update_key, &lightning_output_outcome)
+            .await
+            .expect("Error inserting ContractUpdate");
+
+        let propose_decryption_share_key = ProposeDecryptionShareKey(contract_id);
+        let dec_share: DecryptionShare = Standard.sample(&mut OsRng);
+        let preimage_decryption_share = PreimageDecryptionShare(dec_share);
+        dbtx.insert_new_entry(&propose_decryption_share_key, &preimage_decryption_share)
+            .await
+            .expect("Error insert ProposeDecryptionShare");
+
+        let agreed_decryption_share_key = AgreedDecryptionShareKey(contract_id, 0.into());
+        dbtx.insert_new_entry(&agreed_decryption_share_key, &preimage_decryption_share)
+            .await
+            .expect("Error inserting AgreedDecryptionShareKey");
+
+        let lightning_gateway_key = LightningGatewayKey(pk);
+        let gateway = LightningGateway {
+            mint_channel_id: 100,
+            mint_pub_key: pk.x_only_public_key().0,
+            node_pub_key: pk,
+            api: Url::parse("http://example.com")
+                .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
+            route_hints: vec![],
+            valid_until: SystemTime::now(),
+        };
+        dbtx.insert_new_entry(&lightning_gateway_key, &gateway)
+            .await
+            .expect("Error inserting LightningGateway");
+
+        dbtx.commit_tx()
+            .await
+            .expect("Error committing to database");
+        Ok(())
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn prepare_migration_snapshots() {
+        if let Ok(parent_dir) = env::var("FM_TEST_DB_BACKUP_DIR") {
```

Comment:
We should move this to a function that will generate the destination, prepare an empty db etc. that just takes the name of the snapshot and a function that will populate it.

Reply:
Done